### DISCUSSION TITLE
Unable to use "All Queries" with read-only replica

### DISCUSSION THREAD

**Discussion by gisnicholas** [DaxStudio/DaxStudio#1232] [Q&A]
This is more of a question than a bug report.

For a "scale-out" enabled Premium dataset, I'm only able to use "All Queries" against the read-write replica, but this has no query activity on it. The query activity is on the read-only replica. But when connecting to the read-only replica using the powerbi://...?readonly parameter, the "All Queries" option fails with the following error:

"This operation is not allowed by the Azure Analysis Services product as the current context is read-only."

Is there a way to track query activity against the read-only replica?

Thanks.

**Comment by dgosbell** [comment]
To start a trace like All Queries I need to first create a trace then subscribe to it and I'm guessing that currently all create/alter commands are blocked against a readonly replica. 

You could possibly use a Log Analytics workspace to capture the query activity https://learn.microsoft.com/en-us/power-bi/transform-model/log-analytics/desktop-log-analytics-configure
